plugins/gradle/tooling-extension-impl/src/org/jetbrains/plugins/gradle/tooling/builder/ExternalProjectBuilderImpl.groovy

```diff
@@ -471,6 +472,10 @@ class ExternalProjectBuilderImpl implements ModelBuilderService {
     result
   }
 
+  private static boolean isTestSourceSet(Project project, SourceSet sourceSet) {
```

Comment:
With the change all IDEA modules related to custom source sets with related `Test` task defined will get test source folders.
It's not configurable behavior and might confuse some users, the source folder scope change will break other dependent IDEA modules.
And I don't think it's a good idea in general to introduce such additional conventions.

The IDEA gradle import should stick to `idea` plugin configuration as much as possible.
Unfortunately, `idea` plugin doesn't support custom source sets but we are working closely with Gradle folks to add the support.
Before the support will be added to `idea` plugin we can use `idea.testSourceDirs+= file('some-extra-test-dir')` API to mark such 'test' IDEA explicitly, e.g.
```
apply plugin: 'java'
sourceSets {
  integrationTest
}
apply plugin: 'idea'
idea {
  module {
    testSourceDirs += project.sourceSets.integrationTest.java.srcDirs
    testSourceDirs += project.sourceSets.integrationTest.resources.srcDirs
  }
}
```
and use the following check `idea.testSourceDirs.containsAll(customSourceSet.allJava.srcDirs)' to resolve 'test' IDEA modules.